server/app/services/program/ProgramService.java

```diff
@@ -223,27 +223,26 @@ public ImmutableList<ProgramDefinition> getAllProgramDefinitionVersions(long pro
   }
 
   private CompletionStage<ProgramDefinition> syncProgramAssociations(ProgramModel program) {
-    if (isActiveOrDraftProgram(program)) {
+    VersionModel activeVersion = versionRepository.getActiveVersion();
```

Comment:
**suggestion (non-blocking):** Consider putting this in a helper method. If the helper method is called something like `isDraftVersion()`, then no comment is required here.

Reply:
Oh, now I see that you use `maxVersionForProgram` below, so it's good the way it is.

Reply:
Yep exactly :) was thinking about doing that but then it would involve one more db call to do it in the function and to get the maxVersion for below.